File: Week 6 - Python/Readability/readability.py
import math


def main():

    # takes input from user
    text = input("Text: ")

    # declarations
    word_count = 0
    letters = 0
    sentences = 0

    # determines number of words, letters and sentences
    for char in range(len(text)):
        if text[char].isalpha():
            letters += 1

            if text[char + 1] in ['.', '?', '!', ' ', ',']:
                word_count += 1

        if text[char] in ['.', '?', '!']:
            sentences += 1

    # using Coleman-Liau's index algorithm
    index = round(0.0588 * ((letters / word_count) * 100) - 0.296 * ((sentences/word_count) * 100) - 15.8)

    # prints the grade
    if index < 1:
        print("Before Grade 1")

    elif index > 15:
        print("Grade 16+")

    else:
        print(f"Grade {index}")


main()

File: Week 2 - Arrays/Readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string sentence = get_string("Text : ");

    int c = 0, letters = 0, wordcount = 0, s = 0, length = strlen(sentence);

    for (c = 0; c < length; c++)
    {
        if (isalpha(sentence[c]))
        {
            letters++;
        }

        if (isalpha(sentence[c]) && (isspace(sentence[c + 1]) || sentence[c + 1] == 46 || sentence[c + 1] == 33 || sentence[c + 1] == 63
                                     || sentence[c + 1] == 44))
        {
            wordcount++;
        }

        if (sentence[c] == 46 || sentence[c] == 33 || sentence[c] == 63)
        {
            s++;
        }
    }

    float index = 0.0588 * (100 * (float) letters / (float) wordcount) - 0.296 * (100 * (float) s / (float) wordcount) - 15.8;

    if (index < 16 && index >= 0)
    {
        printf("Grade %i\n", (int) round(index));
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }

}
